packages/data-service/README.md

```diff
@@ -0,0 +1,79 @@
+This package is to be extracted to dcos-ui/data-service.
+
+#API v1
+
+## Mediator Pattern
+
+The [mediator pattern](https://en.wikipedia.org/wiki/Mediator_pattern) is used here
+to decouple data handling from the view components by encapsulating the way they
+communicate. 
+
+With this pattern, a component will receive down all the data `(data down)` it needs via props,
+and send `events up` so the mediator can perform mutations (represented here as a simple callback). 
+The key concept here is that both the data provider and the view component have no idea how each other work.
+Only the caller of `renderComponents` understands how it works.  
+
+```js
+  /*
+   * Receives a stream of React.Component and outputs a single components that 
+   * renders the most recevent event from the Observable.
+  */
+  renderComponents(component$: Observable<React.Component>): React.Component;
+```
+
+Here is an example of how it could be used to render a page for `space shuttles`
+that shows which shuttles are on space, and which are ready to be launched.
+
+```js
+// Imagine we manage a launchpad for interplanetary ships
+import { shuttles$, launchShuttle } from "shuttles/streams";
+import { shuttleSchema } from "shuttles/schema";
+import { Space, LaunchPad, LoadingUI, ErrorMessageUI } from "ui/components"
+import { renderComponents } from "data-service"
+
+// The data composition bellow is using only streams, 
+// IF we had graphql, it could look like this
+/*
+  const query = "
```

Comment:
Can you maybe extract the graphql example into a second example, it's kinda hard to comprehend this way. For me everything here is new and I need to sort out what happens, so thinking how it could be with graphql on thr side is too much for me 😇

Reply:
I agree that that makes reading linearly harder. Will think about something. Thanks.

Reply:
a second complete example would be cool, I think that is the easiest to comprehend

Reply:
Ok